tests/CMakeLists.txt

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required(VERSION 3.13 FATAL_ERROR)
+
+project (az_iot_middleware_freertos_ut LANGUAGES C)
+
+enable_testing()
+
+set(AZ_IOT_REPO_ROOT "${PROJECT_SOURCE_DIR}/..")
+
+# GCC flags required via code coverage
+add_compile_options(-fprofile-arcs -ftest-coverage -O0 -g3)
+add_link_options(-fprofile-arcs -ftest-coverage)
+
+# FreeRTOS Kernel include
+include_directories(${AZ_IOT_REPO_ROOT}/libraries/FreeRTOS/FreeRTOS/Source/include)
+
+# port for FreeRTOS
+if(UNIX)
+    include_directories(${AZ_IOT_REPO_ROOT}/libraries/FreeRTOS/FreeRTOS/Source/portable/ThirdParty/GCC/Posix)
+else()
+    message( FATAL_ERROR "Tests only supported in linux." )
+endif()
+
+# Middleware source
+add_library(az_iot_middleware_freertos
+  ${AZ_IOT_REPO_ROOT}/source/azure_iot_provisioning_client.c
+)
+
+target_compile_options(az_iot_middleware_freertos
+    PRIVATE
+    -Werror
```

Comment:
Nit: These feel like valuable flags for building anything, including samples.  (Our old practice of letting samples compile without as many warnings we should get away from.)

Could we make these global somehow?  That'd also help for like letting this build on other platforms, so in some global file
if (GCC) {
  set(COMPILER_WARNINGS -Werror -Wall etc.)
)
else if (MSVC) {
  set(COMPILER_WARNINGS -whateverMSBuildUses
)

Reply:
I will defer it for now as @danewalton is working on cmake

---
In reply to: [607946369](https://github.com/Azure/azure-iot-middleware-freertos/pull/62#discussion_r607946369) [](ancestors = 607946369)

Reply:
yea these will be update later in another PR. I didn't comment on these since these are being tackled elsewhere. This is ad hoc for now